File: loianeTraining/javaBasico/exercicios-aulas-18-19/Ex23.java
public class Ex23 {
    public static void main(String[] args) {
        int[] a = { 2, 4, 6, 8, 2, 12, 14, 16, 18, 1 };

        var cont = 0;
        for (int i = 1; i < a.length; i++) {
            if (a[i] % 2 == 0) {
                cont++;
            }
            if (cont != i) {
                System.out.println("Numero impar detectado");
                break;
            }
        }
    }
}

File: Loiane_Training_C#/Basico/D/Ex23.cs
using System;

class Ex23
{
    static void Main(string[] args)
    {
        Console.WriteLine("Digite um numero");
        int numero = int.Parse(Console.ReadLine());

        int contador = 0;
        bool isPrime = false;
        for (int i = 1; i <= numero; i++)
        {

            string operacoes = "";
            contador = 0;
            isPrime = false;

            if (i == 1)
            {
                isPrime = true;
            }
            else
            {
                for (int j = 1; j <= i; j++)
                {
                    operacoes += " " + j;


                    if (i % j == 0)
                    {
                        contador++;
                    }

                    if (contador == 2 || i == 1)
                    {
                        isPrime = true;
                    }
                    else
                    {
                        isPrime = false;
                    }
                }
            }


            if (isPrime)
            {
                Console.WriteLine($"{i} e Primo");
            }
            else
            {
                Console.WriteLine($"{i} nao e Primo - Operacoes necessarias para achar primo de {i}: {operacoes}");
            }
        }
    }
}